src/main/java/tasks/TaskList.java

```diff
@@ -0,0 +1,160 @@
+package tasks;
+
+import java.time.format.DateTimeParseException;
+import java.util.List;
+
+import duke.DukeException;
+import duke.Storage;
+
+
+/**
+ * A TaskList class that handles all the methods that manipulates
+ * the task list.
+ */
+public class TaskList {
+    private final List<Task> tasks;
+
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public void add(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * The output that will be shown when the input command is list.
+     * 
+     * @param str
+     * @return a String with all the existing tasks shown in a list view. 
+     */
+    public String showList(String str) {
+        for (int i = 0; i < tasks.size(); i++) {
+            str += (i+1) + ". " + tasks.get(i);
+            if (i != tasks.size() - 1) {
+                str += "\n";
+            }
+        }
+        return str;
+    }
+
+    /**
+     * A method to mark the task as done
+     * 
+     * @param str The input command
+     * @return The output that you want to be printed out in the console
+     */
+    public String markDone(String str) {
+        int completedTaskIndex = Integer.parseInt(str.substring(5)) - 1;
+        tasks.get(completedTaskIndex).taskDone();
+        Storage.getAllTasks(tasks);
+        return "Nice! I've marked this task as done: \n" + tasks.get(completedTaskIndex);
+    }
+
+    /**
+     * A method to create & add a to-do task
+     * 
+     * @param str The input command
+     * @return A response from the bot to indicate a to-do task has been created
+     * and added to the task list
+     */
+    public String todoTask(String str) throws DukeException {
```

Comment:
Maybe you could consider naming this method createTodoTask so the function is clearer